server/src/jobs/auto-discovery.cpp

```diff
@@ -0,0 +1,275 @@
+            powerLink.source = iname;
+            powerLink.link_type = fty::convert<uint32_t>(std::stoi(zconfig_get(link, "type", "1")));
+            if (!(powerLink.source.value() == "0")) {
+                defaultValuesLinks.append(powerLink);
+            }
+        }
+    }
+    logDebug("defaultValuesLinks size={}", defaultValuesLinks.size());
+
+    // Init default parent TBD: To put in a function ???
+    bool deviceCentricView = false;
+    std::string defaultParent("0");
+    const char *CFG_DISCOVERY_DEFAULT_VALUES_AUX("/defaultValuesAux");
+    section = zconfig_locate(config, CFG_DISCOVERY_DEFAULT_VALUES_AUX);
+    if (section) {
+        for (auto item = zconfig_child(section); item; item = zconfig_next(item)) {
+            std::string configName(zconfig_name(item));
+            // the config API call returns the parent internal name, while fty-proto-t needs to carry the database ID
+            if (configName == "parent") {
+                defaultParent = zconfig_value(item);
+                if (defaultParent.empty()) {
+                   defaultParent == "0";
+                }
+                if (defaultParent != "0") {
+                    deviceCentricView = true;
+                }
+            }
+        }
+    }
+    logDebug("defaultParent={}", defaultParent);
+
+    // Get list of protocols
+    commands::protocols::In in_prot;
+    in_prot.address = in.address;
+    Protocols protocols;
+    auto list_protocols = protocols.getProtocols(in_prot);
+    if (!list_protocols) {
+        logError(list_protocols.error().c_str());
+        return;
+    }
+    std::ostringstream list_str;
+    std::for_each((*list_protocols).begin(), (*list_protocols).end(),
+        [&list_str](const fty::job::Type& type) {
+          list_str << Protocols::getProtocolStr(type) << " ";
+        }
+    );
+    logInfo("Found protocols [ {}]", list_str.str());
+
+    // For each protocols read, get assets list available
+    // Note: stop when found first assets with the protocol in progress
+    for (const auto& prot : *list_protocols) {
+        logInfo("Try with protocol {}: ", Protocols::getProtocolStr(prot));
+
+        commands::assets::In in_asset(in);
+        in_asset.protocol = Protocols::getProtocolStr(prot);
+
+        logInfo("in_asset address={}", in_asset.address);
+
+        commands::assets::Out out_asset;
+        Assets assets;
+        if (auto getAssetsRes = assets.getAssets(in_asset, out_asset)) {
+            logInfo("Found asset with protocol {}", Protocols::getProtocolStr(prot));
+
+            auto getStatus = [deviceCentricView]() -> uint {
+                return deviceCentricView ? static_cast<uint>(fty::AssetStatus::Active) : static_cast<uint>(fty::AssetStatus::Nonactive);
+            };
+
+            // Create asset list
+            fty::disco::MessageBus bus;
+            constexpr const char* agent = "fty-discovery-ng";
+            if (auto init = bus.init(agent); !init) {
```

Comment:
I don't think that you need new message bus instance for each protocol

Reply:
Thanks take into account